File: frontend/src/services/analyticsService.js
import apiClient from "../api/axiosConfig";

const API_URL = "/v1/analytics";

/**
 * Fetches the AI-generated weekly review for the current user.
 * @returns {Promise<AxiosResponse<string>>} A promise that resolves to the API response containing the review text.
 */
const getWeeklyReview = () => {
  return apiClient.get(`${API_URL}/weekly-review`);
};

const analyticsService = {
  getWeeklyReview,
};

export default analyticsService;

File: backend/src/main/java/com/risencore/risencore_api/service/AnalyticsService.java
package com.risencore.risencore_api.service;

public interface AnalyticsService {
    /**
     * Generates a weekly review summary for the current user using an AI service.
     *
     * @return A string containing the AI-generated weekly review.
     */
    String generateWeeklyReview();
}
